### DISCUSSION TITLE
"Something broke" when submitting to publish

### DISCUSSION THREAD

**Discussion by waldingl** [nathanwoulfe/Plumber-2#105] [Q&A] [answered]
Nathan

I have created an approval group with and added an admin to it. I then attempted to publish a change as an editor.

When I add a description and a date (current), I get an error saying something broke. The condition is set to invoke the approval group based on one particular field which is the only field that I have attempted to modify as a proof of concept.

Any help is appreciated. We licensed Plumber recently.

Thanks

**Comment by nathanwoulfe** [comment]
Damn my crappy error message! Anything useful in the logs?

Will have a look at this as soon as I can.

**Reply by waldingl** [reply]
Exception

System.NullReferenceException: Object reference not set to an instance of an object.
   at Plumber.Core.Processes.WorkflowProcess.<>c__DisplayClass55_0.<SetApprovalGroup>b__4(PropertyValue v)
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at Plumber.Core.Processes.WorkflowProcess.SetApprovalGroup(WorkflowTaskPoco taskInstance, Int32 nodeId, Int32 initialId)
   at Plumber.Core.Processes.WorkflowProcess.InitiateWorkflow(Int32 nodeId, Int32 authorUserId, String authorComment, String scheduledDate, Nullable`1 attachmentId, String variant)
   at Plumber.Web.Api.ActionsController.InitiateWorkflow(InitiateWorkflowModel model)

**Reply by nathanwoulfe** [reply]
Few more bits to help me track this down:
- new or existing (published) node?
- do you have a default new node workflow configured?
- does the node have a content-type workflow assigned?
- how many stages in the workflow?

Trying to pinpoint where the error is thrown, difficult without line numbers in the exception 😄

**Reply by waldingl** [reply]
Morning

1. Existing node.
2. The default workflow for this node is set to the approval group I created with a single condition as discussed in my original post above.
3. I'm not sure what you mean by a content-type workflow - the only thing I can assign to a group is a document type workflow, which is what you're probably referring to?
4. One stage.

So it appears that it is localised to the Global Content node which is bizarre?

Having said all of that I was able to get Plumber to work on the Home node as an experiment. But the issue with the Home node is none of the Home descendants initiate the workflow just Home itself? Ideally I would like to set Home and everything within it to trigger our compliance workflow.

Progress!

**Reply by nathanwoulfe** [reply]
Cool, will see what I can find. 

Setting on the home node (in the content tree, not the document type) should inherit to all descendants, until a node has permissions set. 

Content/document type permissions do not inherit, so permissions are applied in this order:
 - explicitly set on the current node
 - document type
 - inherited from ancestor

**Reply by waldingl** [reply]
Okay thanks.
I managed to work around the workflow inheritance, that wasn't obvious. I'll wait to hear back from you re: the original issue.

**Comment by nathanwoulfe** [comment]
Pretty sure I've tracked this one down - issue was caused by my assumption that invariant properties would have the default culture, when in fact the culture is null. Will test a bit more, and get a beta out once I'm happy it's resolved.

**Comment by nathanwoulfe** [comment, marked as answer]
I believe this was resolved in v2.0.2
